### ISSUE TITLE
MAMBA_EXE set to `micromamba.bkup (deleted)` during `micromamba self-update`

### ISSUE THREAD

**Issue by hadim** [mamba-org/mamba#2445]
### Troubleshooting docs

- [X] My problem is not solved in the Troubleshooting docs

### Search tried in issue tracker

micromamba.bkup (deleted)

### Latest version of Mamba

- [X] My problem is not solved with the latest version

### Tried in Conda?

Not applicable

### Describe your issue

I noticed that issue since `self-update` exists.

When performing `micromamba self-update` the mamba block in my `bashrc` file set `MAMBA_EXE` to `micromamba.bkup (deleted)` instead of `micromamba`.

Here is the full log:

```bash
$ m self-update
conda-forge/linux-64                                        Using cache
conda-forge/noarch                                          Using cache

  Installing micromamba version: 1.4.2 (currently installed 1.4.1)
  Fetching micromamba from https://conda.anaconda.org/t/*****/conda-forge/linux-64/micromamba-1.4.2-0.tar.bz2


Transaction starting
micromamba                                           5.2MB @  15.5MB/s  0.3s
Download only - packages are downloaded and extracted. Skipping the linking phase.

Reinitializing all previously initialized shells

Modifying RC file "/home/hadim/.bashrc"
Generating config for root prefix "/home/hadim/local/micromamba"
Setting mamba executable to: "/home/hadim/.local/bin/micromamba.bkup (deleted)"
Adding (or replacing) the following in your "/home/hadim/.bashrc" file

# >>> mamba initialize >>>
# !! Contents within this block are managed by 'mamba init' !!
export MAMBA_EXE="/home/hadim/.local/bin/micromamba.bkup (deleted)";
export MAMBA_ROOT_PREFIX="/home/hadim/local/micromamba";
__mamba_setup="$("$MAMBA_EXE" shell hook --shell bash --prefix "$MAMBA_ROOT_PREFIX" 2> /dev/null)"
if [ $? -eq 0 ]; then
    eval "$__mamba_setup"
else
    if [ -f "/home/hadim/local/micromamba/etc/profile.d/micromamba.sh" ]; then
        . "/home/hadim/local/micromamba/etc/profile.d/micromamba.sh"
    else
        export  PATH="/home/hadim/local/micromamba/bin:$PATH"  # extra space after export prevents interference from conda init
    fi
fi
unset __mamba_setup
# <<< mamba initialize <<<
```

For now the easy workaround for my is to simply open the `bashrc` file and fix `MAMBA_EXE`.

**Comment by ruben-arts** [comment]
I have the exact same issue going from `1.4.2` to `1.4.3`

**Comment by mikemhenry** [comment]
I had the same problem going from `1.4.3` to `1.5.8` with `fish` as my shell -- should have this been fixed by #2529 or are old micromamaba updates expected to break? I can open a new issue if this is unexpected @JohanMabille
